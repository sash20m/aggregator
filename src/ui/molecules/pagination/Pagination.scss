.pagination {
    &__arrow-btn {
        width: 30px;
        height: 30px;
        color: white;
        background-color: #467bbd;
        border-radius: 4px;
        border: none;
        outline: none;
        margin: 3px;

        &:hover {
            cursor: pointer;
        }
    }

    &__page-btn {
        width: 30px;
        height: 30px;
        background-color: white;
        color: #467bbd;
        border-radius: 4px;
        border: none;
        outline: none;
        margin: 3px;
        border: 1px solid #9aabc2;

        &:hover {
            cursor: pointer;
        }

        &--active {
            width: 30px;
            height: 30px;
            background-color: #467bbd;
            color: white;
            border-radius: 4px;
            border: none;
            outline: none;
            margin: 3px;

            &:hover {
                cursor: pointer;
            }
        }
    }
}