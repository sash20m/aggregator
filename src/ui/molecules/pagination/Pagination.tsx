import React, { useState, useCallback, useEffect } from 'react';

import './Pagination.scss';

interface Props {
  pages: number;
  changePage(newPage: string): void;
}

export const Pagination: React.FC<Props> = ({ pages, changePage }) => {
  const [pagesArray, setPagesArray] = useState([1, 2, 3, 4, 5, 6, 7]);
  const [firstPage, setFirstPage] = useState<number>(1);
  const [clickedPage, setClickedPage] = useState<number>(1);

  useEffect(() => {
    if (pages < pagesArray.length) setPagesArray(pagesArray.splice(0, pages));
  }, [pages, pagesArray]);

  const goOnePageBack = () => {
    if (firstPage === 1 || pages < 7) return;
    setFirstPage((prevState: number) => prevState - 1);
  };

  const goOnePageFuther = () => {
    if (firstPage + 6 === pages || pages < 7) return;
    setFirstPage((prevState: number) => prevState + 1);
  };

  const goToPage = useCallback(
    (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
      setClickedPage(parseInt(e.currentTarget.name, 10));
      changePage(e.currentTarget.name);
    },
    [changePage]
  );

  return (
    <div className="pagination">
      <button
        className="pagination__arrow-btn"
        type="button"
        onClick={goOnePageBack}
        data-testid="goOnePageBack"
      >
        {'<'}
      </button>
      {pagesArray.map((page, key) => (
        <button
          data-testid={key}
          key={page}
          className={
            firstPage + key === clickedPage
              ? 'pagination__page-btn--active'
              : 'pagination__page-btn'
          }
          type="button"
          name={(firstPage + key).toString(10)}
          onClick={goToPage}
        >
          {firstPage + key}
        </button>
      ))}
      <button
        className="pagination__arrow-btn"
        type="button"
        onClick={goOnePageFuther}
        data-testid="goOnePageFurther"
      >
        {'>'}
      </button>
    </div>
  );
};
