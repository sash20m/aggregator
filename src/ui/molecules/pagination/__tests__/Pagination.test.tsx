import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { Pagination } from '../Pagination';

afterEach(cleanup);

describe('pagination', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <Pagination pages={10} changePage={jest.fn()} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should go one page back', () => {
    const { getByTestId } = render(
      <Pagination pages={10} changePage={jest.fn()} />
    );

    fireEvent.click(getByTestId('goOnePageBack'));

    expect(getByTestId('0')).toHaveTextContent('1');
  });

  it('should go one page futher', () => {
    const { getByTestId } = render(
      <Pagination pages={10} changePage={jest.fn()} />
    );

    fireEvent.click(getByTestId('goOnePageFurther'));

    expect(getByTestId('0')).toHaveTextContent('2');
  });

  it('current page should be active', () => {
    const { getByTestId } = render(
      <Pagination pages={10} changePage={jest.fn()} />
    );

    fireEvent.click(getByTestId('1'));

    expect(getByTestId('1')).toHaveClass('pagination__page-btn--active');
  });
});
