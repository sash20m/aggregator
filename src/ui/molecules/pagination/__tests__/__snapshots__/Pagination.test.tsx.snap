// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <button
      class="pagination__arrow-btn"
      data-testid="goOnePageBack"
      type="button"
    >
      &lt;
    </button>
    <button
      class="pagination__page-btn--active"
      data-testid="0"
      name="1"
      type="button"
    >
      1
    </button>
    <button
      class="pagination__page-btn"
      data-testid="1"
      name="2"
      type="button"
    >
      2
    </button>
    <button
      class="pagination__page-btn"
      data-testid="2"
      name="3"
      type="button"
    >
      3
    </button>
    <button
      class="pagination__page-btn"
      data-testid="3"
      name="4"
      type="button"
    >
      4
    </button>
    <button
      class="pagination__page-btn"
      data-testid="4"
      name="5"
      type="button"
    >
      5
    </button>
    <button
      class="pagination__page-btn"
      data-testid="5"
      name="6"
      type="button"
    >
      6
    </button>
    <button
      class="pagination__page-btn"
      data-testid="6"
      name="7"
      type="button"
    >
      7
    </button>
    <button
      class="pagination__arrow-btn"
      data-testid="goOnePageFurther"
      type="button"
    >
      &gt;
    </button>
  </div>
</DocumentFragment>
`;
